### PR TITLE
deno 1.24.1

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Needs tinycc's `libtcc.a` now. deno_ffi build.rs builds a copy from a submodule and uses it there

https://github.com/denoland/deno/pull/15139

**Comment by aslilac** [comment]
Deno 1.24.0 was released today, so we could update and get rid of that TCC_PATH patch

**Comment by SMillerDev** [comment]
At this point it might be better for you to make a new PR @aslilac

**Comment by eserozvataf** [comment]
:/

**Comment by cowboyd** [comment]
Anything I can do to unblock this? I'm currently using homebrew to manage deno, so don't really want to go through the hassle of changing the installation method.

**Comment by aslilac** [comment]
> Anything I can do to unblock this? I'm currently using homebrew to manage deno, so don't really want to go through the hassle of changing the installation method.

You can try to help figure out what's broken in the Linux build. It's not immediately obvious to me and I haven't gotten to spend much time investigating yet.

**Comment by SMillerDev** [comment]
```
     = note: "cc" "-m64" "/tmp/rustc5PQxqN/symbols.o" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/deps/deno-14973ab5e6fc5a9e.deno_emit-cb00938170714bf5.deno_emit.e0113723-cgu.0.rcgu.o.rcgu.o" "-Wl,--as-needed" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/deps" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/gn_out/obj/" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/build/ring-168ebeefdf436a94/out" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/tinycc" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/build/libffi-sys-a833a09557457c94/out/libffi-root/lib" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/build/libffi-sys-a833a09557457c94/out/libffi-root/lib64" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/build/libsqlite3-sys-e9e4dbdcee54d22d/out" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/build/lzzzz-62edba9c60fbabf1/out" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/build/sys-info-0f9a0f6edacfc691/out" "-L" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/build/zstd-sys-005af013a83b5d16/out" "-L" "/home/linuxbrew/.linuxbrew/Cellar/rust/1.62.1/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/tmp/rustc5PQxqN/libzstd_sys-fcaaf12801ffa07d.rlib" "/tmp/rustc5PQxqN/libsys_info-426cca129dd8f721.rlib" "/tmp/rustc5PQxqN/liblzzzz-8435d57df8534352.rlib" "/tmp/rustc5PQxqN/liblibsqlite3_sys-8effc812c37ef020.rlib" "/tmp/rustc5PQxqN/libdeno_ffi-f225b36713d8af0f.rlib" "/tmp/rustc5PQxqN/liblibffi_sys-501fae4c3bc0e3b7.rlib" "/tmp/rustc5PQxqN/libring-e12c454eb5e6abe7.rlib" "/tmp/rustc5PQxqN/libv8-ff2316ad0b39db24.rlib" "-Wl,--start-group" "-Wl,--end-group" "/home/linuxbrew/.linuxbrew/Cellar/rust/1.62.1/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-5f437c82c3555c7b.rlib" "-Wl,-Bdynamic" "-ldl" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-znoexecstack" "-L" "/home/linuxbrew/.linuxbrew/Cellar/rust/1.62.1/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/tmp/deno-20220721-15418-13wdws3/deno/target/release/deps/deno-14973ab5e6fc5a9e" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro,-znow" "-nodefaultlibs"
    = note: /home/linuxbrew/.linuxbrew/bin/ld: /tmp/rustc5PQxqN/libdeno_ffi-f225b36713d8af0f.rlib(libtcc.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIE
            /home/linuxbrew/.linuxbrew/bin/ld: failed to set dynamic section sizes: bad value
            collect2: error: ld returned 1 exit status
            
  
  error: could not compile `deno` due to previous error
```

**Comment by aslilac** [comment]
Oh wait, do we just need to add `-fPIE` to the tinycc build? I'll try to play around with it over the weekend

**Comment by zachauten** [comment]
Deno v1.24.1 was released yesterday. I opened PR #106887 for it, but I can close it if @branchvincent wants to update this PR, since there's a lot of discussion already here.

**Comment by aslilac** [comment]
> Deno v1.24.1 was released yesterday. I opened PR #106887 for it, but I can close it if @branchvincent wants to update this PR, since there's a lot of discussion already here.

It doesn't look like you made any changes to address the failing build. 😅 I appreciate the enthusiasm, but anyone can bump a version number, it's the failing build that is blocking things.

**Comment by branchvincent** [comment]
> Oh wait, do we just need to add `-fPIE` to the tinycc build

I think so, let's try it

File: Formula/deno.rb [hunk 1]
```diff
@@ -98,6 +105,15 @@ def install
       system "ninja", "-C", "out"
     end
 
+    resource("tinycc").stage buildpath/"tinycc"
+    cd "tinycc" do
+      ENV.append_to_cflags "-fPIE" if OS.linux?
```
Comment (SMillerDev) [inline Formula/deno.rb:110]: AFAIK this means it's doing a static build instead of shared

**Comment by aslilac** [comment]
Yay! :partying_face: looks like that worked!

File: Formula/deno.rb (hunk 1 above)
Comment (carlocab) [reply Formula/deno.rb:110]: Yes, vendors its own `libtinycc`. Not cool.

**Review by carlocab — APPROVED** [review]

**Comment by branchvincent** [comment]
@aslilac thanks for your help here!

**Pull request closed without merging**